apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+    var carrierTag = document.getElementById('contact-carrier');
+    var threadMessages = this.threadMessages;
+    var number = thread.participants[0];
+    var phoneDetails;
+    var address;
+
+    // The carrier banner is meaningless and confusing in
+    // group message mode.
+    if (thread.participants.length === 1 &&
+        (contacts && contacts.length)) {
+
+      address = Settings.supportEmailRecipient && Utils.isEmailAddress(number) ?
+                contacts[0].email : contacts[0].tel;
+
+      phoneDetails = Utils.getPhoneDetails(number, address);
+
+      if (phoneDetails) {
+        carrierTag.innerHTML = SharedComponents.phoneDetails(
+          phoneDetails
+        ).toString();
+
+        threadMessages.classList.add('has-carrier');
+      } else {
+        threadMessages.classList.remove('has-carrier');
+      }
+    } else {
+      // Hide carrier tag in group message or unknown contact cases.
+      threadMessages.classList.remove('has-carrier');
+    }
+  },
+
+  // Method for updating the header with the info retrieved from Contacts API
+  updateHeaderData: function conv_updateHeaderData() {
+    var thread, number;
+
+    thread = Threads.active;
+
+    if (!thread) {
+      return Promise.resolve();
+    }
+
+    number = thread.participants[0];
+
+    // Add data to contact activity interaction
+    this.headerText.dataset.number = number;
+
+    return Contacts.findByAddress(number).then((contacts) => {
+      // For the basic display, we only need the first contact's information
+      // e.g. for 3 contacts, the app displays:
+      //
+      //    Jane Doe (+2)
+      //
+      var details = Utils.getContactDetails(number, contacts);
+      // Bug 867948: contacts null is a legitimate case, and
+      // getContactDetails is okay with that.
+      var contactName = details.title || number;
+      this.headerText.dataset.isContact = !!details.isContact;
+      this.headerText.dataset.title = contactName;
+
+      var headerContentTemplate = thread.participants.length > 1 ?
+        this.tmpl.groupHeader : this.tmpl.header;
+      this.setHeaderContent({
+        html: headerContentTemplate.interpolate({
+          name: contactName,
+          participantCount: (thread.participants.length - 1).toString()
+        })
+      });
+
+      this.updateCarrier(thread, contacts);
+    });
+  },
+
+  /**
+   * Updates header content since it's used for different panels and should be
+   * carefully handled for every case. In Thread panel header contains HTML
+   * markup to support bidirectional content, but other panels still use it with
+   * mozL10n.setAttributes as it would contain only localizable text. We should
+   * get rid of this method once bug 961572 and bug 1011085 are landed.
+   * @param {string|{ html: string }|{id: string, args: Object }} contentL10n
+   * Should be either safe HTML string or l10n properties.
+   * @public
+   */
+  setHeaderContent: function conv_setHeaderContent(contentL10n) {
```

Comment:
yeah, main use of this method was to be called by other panels... and we don't want that :)